packages/block-library/src/embed/index.js

```diff
@@ -19,35 +24,37 @@ export const settings = getEmbedBlockSettings( {
 	icon: embedContentIcon,
 	// Unknown embeds should not be responsive by default.
 	responsive: false,
-	transforms: {
-		from: [
-			{
-				type: 'raw',
-				isMatch: ( node ) => node.nodeName === 'P' && /^\s*(https?:\/\/\S+)\s*$/i.test( node.textContent ),
-				transform: ( node ) => {
-					return createBlock( 'core/embed', {
-						url: node.textContent.trim(),
-					} );
-				},
-			},
-		],
-	},
+	transforms,
 } );
 
 export const common = commonEmbeds.map(
 	( embedDefinition ) => {
+		const embedSettings = getEmbedBlockSettings( embedDefinition.settings );
 		return {
 			...embedDefinition,
-			settings: getEmbedBlockSettings( embedDefinition.settings ),
+			settings: {
+				...embedSettings,
+				transforms: {
+					from: concat( get( transforms, 'from', [] ), get( embedSettings, 'transforms.from', [] ) ).filter( Boolean ),
+					to: concat( get( transforms, 'to', [] ), get( embedSettings, 'transforms.to', [] ) ).filter( Boolean ),
+				},
```

Comment:
Basically, apply combined transformations of `core/embed` as well as other `core-embed/common/*`